tools/testbench/README.md

```diff
@@ -105,50 +172,83 @@ octave -q --eval "pkg load signal io; [n_fail]=process_test('eq-iir', 32, 32, 48
+sox to raw 32 bit 48 kHz stereo format, then processed by testbench,
+converted back to wav.
 
 ```
 cd $SOF_WORKSPACE/sof
-sox --encoding signed-integer /usr/share/sounds/alsa/Front_Center.wav -L -r 48000 -c 2 -b 32 in.raw
-tools/testbench/build_testbench/install/bin/testbench -r 48000 -R 48000 -c 2 -n 2 -b S32_LE -p 1,2 \
-   -t tools/build_tools/topology/topology2/development/sof-hda-benchmark-dcblock32.tplg \
+scripts/rebuild-testbench.sh
+scripts/build-tools.sh -t
+sox --encoding signed-integer /usr/share/sounds/alsa/Front_Left.wav -L -r 48000 -c 2 -b 32 in.raw
+tools/testbench/build_testbench/install/bin/sof-testbench3 -r 48000 -R 48000 -c 2 -n 2 -b S32_LE \
+   -t tools/build_tools/test/topology/test-playback-ssp5-mclk-0-I2S-dcblock-s32le-s32le-48k-24576k-codec.tplg \
    -i in.raw -o out.raw
 sox --encoding signed-integer -L -r 48000 -c 2 -b 32 out.raw out.wav
 aplay out.wav
 ```
 
-The difference in command line is the use of other topology for IPC4
-and use of -p command line option to select pipelines 1 and 2 from
-this topology for scheduling and connecting to input and output files.
-In this topology the host playback pipeline is 1 and the dai playback
-side pipeline is 2. Capture side pipelines would be 3 and 4. And
-running both playback and capture would need more input and output
-files to be added to -i and -o as comma separated. The more advanced
-use cases simulations with IPC4 are still under work.
+The testbench binary can be debugged and profiled with native
+executable tools like gdb, ddd (GUI for gdb), gprof, and valgrind.
+
+### Profiling of testbench run on xtensa
+
+Fist, the testbench is build in this example for Intel MTL
+platform. This needs access to Cadence Xplorer toolchain with the core
+confiration for the platform.
+
+```
+cd $SOF_WORKSPACE/sof
+export XTENSA_TOOLS_ROOT=~/xtensa/XtDevTools
+export ZEPHYR_TOOLCHAIN_VARIANT=xt-clang
+scripts/rebuild-testbench.sh -p mtl
+source tools/testbench/build_xt_testbench/xtrun_env.sh
+```
+
+Next the testbench is run with xt-run simulator. The example component
+is dynamic range processing (DRC). For proling data generate use
+option --profile. Note that this is a lot slower than native run. An
+1s extract of a pink noise file is used as example.
+
+```
+sox --encoding signed-integer /usr/share/sounds/alsa/Noise.wav -L -r 48000 -c 2 -b 32 in.raw trim 0.0 1.0
+$XTENSA_PATH/xt-run --profile=profile.out tools/testbench/build_xt_testbench/testbench \
+   -q -r 48000 -R 48000 -c 2 -n 2 -b S32_LE \
+   -t tools/build_tools/test/topology/test-playback-ssp5-mclk-0-I2S-drc-s32le-s32le-48k-24576k-codec.tplg \
+   -i in.raw -o out.raw 2> trace.txt
+```
+
+Then convert the profiler data to readable format and check it.
 
-The debugging and profiling of IPC4 testench is similar as with IPC3.
+Note: Current version of testbench does not have functional quiet mode
+"-q" switch to suppress trace. Due all debug traces print
+out. Majority of performance is spent in printing.
```

Comment:
"Due all debug traces print out." is a bit unclear, what do you mean by this? Also "Majority of performance" sounds a bti off, "Majority of cycles" ?

Reply:
Thanks, oops, what did I write. Also the -q option was removed as duplicate of -d , need to fix this.

Reply:
Fixed now!